src/Forum/ForumServiceProvider.php

```diff
@@ -141,6 +147,25 @@ protected function populateRoutes(RouteCollection $routes)
         );
     }
 
+    protected function checkCustomLessFormatBeforeSaving()
+    {
+        $this->app->make('events')->listen(PrepareSerializedSetting::class, function (PrepareSerializedSetting $event) {
+            if ($event->key === 'custom_less') {
+                $parser = new Less_Parser();
+
+                try {
+                    // Check the custom less format before saving
+                    // variables names are not checked, we would have to set them and call getCss() to check them
```

Comment:
So what happens if you do have core variables in your custom LESS? Does it fail validation?

Reply:
The `parse` method does not handle variables at all apparently, it only checks and parse format. So you can input any variable name without an exception being thrown.

If we were to call $parser->getCss() afterwards, that part would try to fill variables and fail if their name is invalid.

Would it be worth it to also load and check variables ? I'm not sure if the part injecting variables has been designed to be reusable

Reply:
I see, that makes sense.

Hmm I think ideally when you save custom LESS it should attempt to recompile the forum CSS straight away, rather than doing it on the next forum page load. That way we can be sure that the custom LESS is valid in the context of all core/extension LESS (and variables).

That will obviously take some bigger refactoring to do though...

Reply:
Indeed :grin: 

But it's tricky. If the check is made before saving the custom less, it's not easy to push the new value into the WebApp just for testing. Maybe by pushing another class that implements the `SettingsRepositoryInterface` interface when creating the WebApp ? The value is fetched directly from the settings https://github.com/flarum/core/blob/40ebc13292b64271418aa3d5683d0927e6db778e/src/Forum/WebApp.php#L55

And if the check is done after... I have no idea how to rollback in case of exception :sweat_smile: